draft-sharabayko-mops-srt.md

```diff
@@ -2870,7 +2870,7 @@ On the protocol control level, SRT does not encrypt packet headers.
 Therefore it has some vulnerabilities similar to TCP {{RFC6528}}:
 
 - a peer tells a counterpart its public IP during the handshake that is visible to any attacker,
-- an attacker may count the number of systems behind a Network Address Translator (NAT) by establishing a number of SRT connections and identifying the number of different sequence number "spaces", depending on the implementation of the initial sequence number generation,
+- an attacker may potentially count the number of systems behind a Network Address Translator (NAT) by establishing a number of SRT connections and identifying the number of different sequence number "spaces". The possibility of attack depends on the implementation of the initial sequence number (ISN) generation,
```

Comment:
This statement is unclear to me. What are the "systems" an attacker might count? What are "sequence number spaces", and how does knowing their number help the attacker?

Reply:
Sequence number spaces - something Maxim has mentioned in TCP documentation as well. According to him, this term should be recognisable by the reader. But I've had exactly the same question. The meaning is the following:
- for each SRT packet we have sequence number which is growing by 1 from packet to packet,
- right after connection we randomly choose the start sequence number (e.g., 1660778620) and the very first packet gets this number. The next packet will be with 1660778621, the next 1660778622, etc.
- on the other hand, for any other connection, this very first randomly chosen sequence number can be 255222121, the next packet will get 255222122, the next 255222123, etc.
- By spaces Maxim assumes 166... and 255... sets of numbers. And the possible attack is related to identifying theses spaces because behind them there is a logic of increasing sequence number by 1 each time you send a packet.

Reply:
@maxsharabayko What do you mean by systems here?

Reply:
It is a bit opportunistic maybe, but an attacker can count the number of SRT **processes** by tracking the ranges of SRT Socket IDs. As in our implementation, a library is instantiated and establishes the first connection generating a random Socket ID. A further connection gets consecutive SRT Socket IDs.
So one process will generate SRT Socket IDs 126, 127, 128, ... While another process will generate 5867, 5868, 5869, ...
If we assume one system  (e.g. Makito) runs only one SRT process, then an attacker can estimate the number of systems behind a NAT.

I actually inaccurately referred to ISN in the paragraph. We generate ISN randomly for each connection, so it is impossible to use it to track anyone. But if some other implementation does not do that, then it is also a vulnerability give that no SRT packet headers are encrypted.

Reply:
```suggestion
- an attacker may potentially count the number of SRT processes behind a Network 
Address Translator (NAT) by establishing multiple SRT connections and tracking 
the ranges of SRT Socket IDs. If a random Socket ID is generated for the first 
connection, subsequent connections may get consecutive SRT Socket IDs. Assuming 
one system runs only one SRT process, for example, then an attacker can estimate 
the number of systems behind a NAT.

- similarly, the possibility of attack depends on the implementation of the initial 
sequence number (ISN) generation. If an ISN is not generated randomly for each 
connection, an attacker may potentially count the number of systems behind a 
Network Address Translator (NAT) by establishing a number of SRT connections and 
identifying the number of different sequence number "spaces", given that no SRT 
packet headers are encrypted.
```

Reply:
I couldn't write better! 👍